internal/service/directconnect/macsec_key.go

```diff
@@ -0,0 +1,173 @@
+package directconnect
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceMacSecKey() *schema.Resource {
+	return &schema.Resource{
+		// MacSecKey resource only supports create (Associate), read (Describe) and delete (Disassociate)
+		Create: resourceMacSecKeyCreate,
+		Read:   resourceMacSecKeyRead,
+		// You cannot modify a MACsec secret key after you associate it with a connection.
+		// To modify the key, disassociate the key from the connection, and then associate
+		// a new key with the connection
+		Update: schema.Noop,
```

Comment:
Remove this and set all the attributes that are not `Computed`-only as `ForceNew: true`.